### PR TITLE
fix(docker): Workaround for rootless containers as git operations may fail due to dubious ownership of /src

### PR DESCRIPTION
**Pull request by bmahe**
PR checklist:

- [X] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [X] Tests included or PR comment includes a reproducible test plan
- [X] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [X] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

**Comment by bmahe** [comment]
Here is how I tested it:
```bash
[bruno@buo]$ podman run --rm -v "${PWD}:/src" --security-opt "label=disable"  -it localhost/semgrep:latest /bin/bash -c "cd /src && git status" | head -n 4
On branch service
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
```

```bash
[bruno@buo]$ podman run --rm -v "${PWD}:/src" --security-opt "label=disable" --user semgrep  -it localhost/semgrep:latest /bin/bash -c "cd /src && git status" | head -n 4
On branch service
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
```

Prior to this PR, one can observe the failure:
```bash
[bruno@buo]$ podman run --rm -v "${PWD}:/src" --security-opt "label=disable" --user semgrep  -it returntocorp/semgrep:develop /bin/bash -c "cd /src && git status"
fatal: detected dubious ownership in repository at '/src'
To add an exception for this directory, call:

        git config --global --add safe.directory /src
```

**Review by IagoAbal — APPROVED** [review]
Just stamping.

File: Dockerfile [hunk 1]
```diff
@@ -190,6 +190,12 @@ RUN addgroup --system semgrep \
 # We can set it by default once we fix the circle ci workflows
 #USER semgrep
 
+# Workaround for rootless containers as git operations may fail due to dubious ownership of /src
+RUN printf "[safe]\n	directory = /src"  > ~root/.gitconfig
+RUN printf "[safe]\n	directory = /src"  > ~semgrep/.gitconfig && \
+	chown semgrep:semgrep ~semgrep/.gitconfig
```
Comment (tpetr) [inline Dockerfile:196]: any reason we shouldn't globally mark `/src` as a safe directory?

Comment (bmahe) [reply Dockerfile:196]: Not sure to follow the question. Can you rephrase it?
Globally how?

Comment (tpetr) [reply Dockerfile:196]: Sure, I mean mark `/src` as a safe directory for all users (not just `root` and `semgrep`). You can do this by running `git config --global --add safe.directory /src` or by editing `/etc/gitconfig`

Comment (bmahe) [reply Dockerfile:196]: Ah great idea!
We are also observing a lot of variation in ways /src is mounted across the various container platforms and technologies. So I would be curious to explore that a bit more and see if there are other alternatives to makring /src as a safe directory
